### PR TITLE
GEN-1267 - refact(CarTrialExtensionBlock): factor out "render banner"

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

- Factor out "render banner" logic in a hook.

## Justify why they are needed

Makes `CarTrialExtensionBlock` easier to understand/maintain.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3389** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3389" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3389?utm_source=stack-comment).

File: apps/store/src/features/carDealership/CarTrialExtensionBlock.tsx [hunk 1]
```diff
@@ -101,6 +78,48 @@ const useCarTrialQuery = (params: UseCarTrialQueryParams) => {
...
+
+const useAddNotificationBanner = () => {
+  const { addBanner } = useGlobalBanner()
+  const { dateFull } = useFormatter()
+  const { t } = useTranslation('carDealership')
+
+  const addNotificationBanner = useCallback(
+    ({ data, requirePaymentConnection = false }: AddNotificationBannerOptions) => {
+      if (requirePaymentConnection) {
+        const today = new Date()
```
Comment (alebedev) [inline apps/store/src/features/carDealership/CarTrialExtensionBlock.tsx:94]: This date manipulation feels a bit low-lever, can we extract a helper function here?

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/features/carDealership/CarTrialExtensionBlock.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/features/carDealership/CarTrialExtensionBlock.tsx:94]: Alternatively, can we do it better with `date-fns`, which we already have?

Comment (guilhermespopolin) [reply apps/store/src/features/carDealership/CarTrialExtensionBlock.tsx:95]: Updated

**Pull request merged**